include/SFML/System/Vector2.inl

```diff
@@ -53,6 +53,138 @@ y(static_cast<T>(vector.y))
 }
 
 
+////////////////////////////////////////////////////////////
+template <typename T>
+T Vector2<T>::length() const
+{
+    static_assert(std::is_floating_point_v<T>, "Vector2::length() is only supported for floating point types");
+
+    using std::sqrt; // allow ADL
+
+    return sqrt(lengthSq());
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+T Vector2<T>::lengthSq() const
+{
+    static_assert(std::is_floating_point_v<T>, "Vector2::lengthSq() is only supported for floating point types");
+
+    return this->dot(*this);
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+Vector2<T> Vector2<T>::normalized() const
+{
+    static_assert(std::is_floating_point_v<T>, "Vector2::normalized() is only supported for floating point types");
+
+    assert(*this != Vector2<T>());
+    return (*this) / length();
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+Angle Vector2<T>::signedAngleTo(const Vector2<T>& rhs) const
+{
+    static_assert(std::is_floating_point_v<T>, "Vector2::signedAngleTo() is only supported for floating point types");
+
+    using std::atan2; // allow ADL
+
+    assert(*this != Vector2<T>());
+    assert(rhs != Vector2<T>());
+    return radians(atan2(this->cross(rhs), this->dot(rhs)));
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+Angle Vector2<T>::polarAngle() const
+{
+    static_assert(std::is_floating_point_v<T>, "Vector2::polarAngle() is only supported for floating point types");
+
+    using std::atan2; // allow ADL
+
+    assert(*this != Vector2<T>());
+    return radians(atan2(y, x));
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+Vector2<T> Vector2<T>::rotatedBy(Angle angle) const
+{
+    static_assert(std::is_floating_point_v<T>, "Vector2::rotatedBy() is only supported for floating point types");
+        
+    using std::sin; // allow ADL
+    using std::cos;
+
+    // No zero vector assert, because rotating a zero vector is well-defined (yields always itself)
+    T c = cos(angle.asRadians());
+    T s = sin(angle.asRadians());
```

Comment:
I've considered a few times whether defining `sf::sin` and `sf::cos` for `sf::Angle` makes sense. It would allow easy ADL usage of `Angle`s and remove many more `asRadians()` calls.

Reply:
I would say that's a good problem to solve once we actually have it 😉 

I'm not even sure if many people use `Vector<double>` for example. And I've never seen anyone use custom types. I completely overengineered that in Thor already, probably for nothing. The ADL-enabled trigonometric functions are likely good enough, if there's need for it at all.

Reply:
Right, not a problem for this PR, just a thought I wanted to share. I might make a PR for that later if I can find good motivating examples to add those extra free functions.

Reply:
Ah, sorry I totally misread. You were talking about `sf::Angle`, I was understanding that we extend the trigonometry, so that user can specialize it for their own types.

A first great addition would actually be `sf::Angle::Pi` and `sf::Angle::Tau`.